libs/hardwareWallet/ledger/ledgerDeviceListener/useLedgerDeviceListener.js

```diff
@@ -8,18 +8,39 @@ import {
   setHWDevices,
 } from '@hardwareWallet/store/actions';
 import { subscribeToLedgerDeviceEvents } from '@libs/hardwareWallet/ledger/ledgerDeviceListener/subscribeToLedgerDevices';
-import { selectHardwareDevices } from '@hardwareWallet/store/selectors/hwSelectors';
-import { getConnectedHWDevices } from '@libs/hardwareWallet/ledger/ledgerLiskAppIPCChannel/clientLedgerHWCommunication';
+import {
+  selectCurrentHWDevice,
+  selectHardwareDevices,
+} from '@hardwareWallet/store/selectors/hwSelectors';
+import {
+  getConnectedHWDevices,
+  getIsInsideLiskApp,
+} from '@libs/hardwareWallet/ledger/ledgerLiskAppIPCChannel/clientLedgerHWCommunication';
 import { usePrevious } from 'src/utils/usePrevious';
 import DeviceToast from '@hardwareWallet/components/DeviceToast/DeviceToast';
 
 export function useLedgerDeviceListener() {
   const dispatch = useDispatch();
   const hwDevices = useSelector(selectHardwareDevices);
+  const currentHwDevice = useSelector(selectCurrentHWDevice);
   const prevHWDevices = usePrevious(hwDevices);
   const [hasCheckedHWDevices, setHasCheckedHWDevices] = useState(false);
   const { ipc } = window;
 
+  // eslint-disable-next-line consistent-return
+  useEffect(() => {
+    async function checkStatus() {
+      const pubKey = await getIsInsideLiskApp(currentHwDevice?.path, currentHwDevice?.accountIndex);
+      const isInsideApp = !!pubKey;
+      dispatch(setCurrentHWDevice({ ...currentHwDevice, isAppOpen: isInsideApp }));
+    }
+
+    if (currentHwDevice?.path) {
+      const id = setInterval(checkStatus, 4000);
```

Comment:
Connect retry const? define the constant and use that variable libs/hardwareWallet/ledger/constants.js